source/code/omi_preexec/omi-mysql.cpp

```diff
@@ -82,113 +83,19 @@ static int MySQL_ConfigurationParser(uid_t uid, gid_t gid, const std::wstring& c
         return 0;
     }
 
-    SCX_LOGHYSTERICAL(hLog, StrAppend(L"Parsing MySQL configuration file: ", cnfFile));
+    SCX_LOGHYSTERICAL(hLog, StrAppend(L"Loading MySQL configuration file: ", cnfFile));
 
-    std::vector<std::wstring> lines;
-    try
+    MySQL::MySQL_ConfigurationScanner configScanner;
+    if (!configScanner.Scan(cnfFile, auth))
```

Comment:
It looks like the Scan() method just handles all of the reading/parsing of the file, handling recursion automatically. Nice design, that's exactly what I would have done.

That also has the benefit of encapsulating all config file handling to one place, which is great.